java/sidecar/src/main/java/com/instaclustr/cassandra/sidecar/operation/OperationTask.java

```diff
@@ -0,0 +1,111 @@
+package com.instaclustr.cassandra.sidecar.operation;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+import com.instaclustr.cassandra.sidecar.model.operation.OperationState;
+import com.instaclustr.cassandra.sidecar.model.operation.Operation;
+import com.instaclustr.cassandra.sidecar.model.result.OperationResult;
+import jmx.org.apache.cassandra.service.StorageServiceMBean;
+
+public abstract class OperationTask<O extends Operation, R extends OperationResult> implements Callable<R> {
+
+    private final O operation;
+
+    private final R result;
+
+    private StopWatch stopWatch = new StopWatch();
+
+    protected StorageServiceMBean storageServiceMBean;
+
+    public OperationTask(final O operation, final R result) {
+        this.operation = operation;
+        this.result = result;
+    }
+
+    @Override
+    public R call() {
+
+        stopWatch.start();
+
+        result.setOperationState(OperationState.RUNNING);
+        result.setStart(stopWatch.getStart());
+        result.setId(operation.getId());
+
+        try {
+            executeTask(operation, result);
+            result.setOperationState(OperationState.FINISHED);
+        } catch (Exception e) {
+            result.setOperationState(OperationState.CRASHED);
+            result.setException(e);
+        }
+
+        stopWatch.stop();
+
+        result.setStop(stopWatch.getStop());
+        result.setDuration(stopWatch.duration().get());
+
+        return result;
+    }
+
+    protected abstract void executeTask(final O operation, final R result) throws Exception;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public R getOperationResult() {
+        return result;
+    }
+
+    public void setStorageServiceMBean(StorageServiceMBean storageServiceMBean) {
+        this.storageServiceMBean = storageServiceMBean;
+    }
+
+    public static class StopWatch {
```

Comment:
Look into using Guava's `Stopwatch` class.

Reply:
That has only `elapsed()` method which gives me Duration, not Instant. I want to catch two points in time and make a duration between them as start, stop and duration is in the result. If this `Stopwatch` gave me `Duration`, I do not know when that duration "started and stopped".